apps/dashboard/src/components/workflow-editor/steps/email/extensions/variable-view.tsx

```diff
@@ -0,0 +1,50 @@
+import { useCallback, useMemo, useState } from 'react';
+import { NodeViewProps } from '@tiptap/core';
+import { NodeViewWrapper } from '@tiptap/react';
+
+import { EditVariablePopover } from '@/components/variable/edit-variable-popover';
+import { parseVariable } from '@/components/primitives/control-input/variable-plugin/utils';
+import { VARIABLE_REGEX_STRING } from '@/components/primitives/control-input/variable-plugin';
+import { VariablePill } from '@/components/variable/variable-pill';
+
+export function VariableView({ node, updateAttributes, editor }: NodeViewProps) {
```

Comment:
This component represents the variable pill plus the editing popover. It's used as a Maily variable extension view.